### PR TITLE
chore: add [convert] extra

### LINKED ISSUE #510: "wheel convert" depends on setuptools
~~Since 0de0c1a, wheel no longer declares a runtime dependency on setuptools. However, the `convert` command still needs setuptools to work.~~ (Never mind this part, there was never a runtime dependency to start with.)

wheel doesn't declare a runtime dependency on setuptools, but the `convert` command needs it to work. Here's what happens when setuptools is not installed:

```console
> .\venv\Scripts\wheel.exe convert
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File ".\venv\Scripts\wheel.exe\__main__.py", line 7, in <module>
  File ".\venv\Lib\site-packages\wheel\cli\__init__.py", line 91, in main
    args.func(args)
  File ".\venv\Lib\site-packages\wheel\cli\__init__.py", line 29, in convert_f
    from .convert import convert
  File ".\venv\Lib\site-packages\wheel\cli\convert.py", line 10, in <module>
    from ..bdist_wheel import bdist_wheel
  File ".\venv\Lib\site-packages\wheel\bdist_wheel.py", line 24, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

### PR DESCRIPTION
**Pull request by henryiii**
Fix #510 by adding an extra for `wheel convert`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,6 +39,9 @@ wheel = "wheel.cli:main"
 bdist_wheel = "wheel.bdist_wheel:bdist_wheel"
 
 [project.optional-dependencies]
+convert = [
+    "setuptools",
```
Comment (chrysle) [inline pyproject.toml:43]: Should this be pinned?

```suggestion
    "setuptools >= 65",
```

**Comment by agronholm** [comment]
I would prefer dropping the setuptools requirement altogether by refactoring the code.

**Comment by agronholm** [comment]
I've started said refactoring now, but this will take time.

**Comment by agronholm** [comment]
Closing in favor of #640.

**Pull request closed without merging**